econml/iv/dr/_dr.py

```diff
@@ -73,83 +246,1246 @@ def _effect_estimate(self, nuisances):
+        return self.ortho_learner_model_final_._original_featurizer
+
+    def cate_feature_names(self, feature_names=None):
+        """
+        Get the output feature names.
+
+        Parameters
+        ----------
+        feature_names: list of strings of length X.shape[1] or None
+            The names of the input features. If None and X is a dataframe, it defaults to the column names
+            from the dataframe.
+
+        Returns
+        -------
+        out_feature_names: list of strings or None
+            The names of the output features :math:`\\phi(X)`, i.e. the features with respect to which the
+            final CATE model for each treatment is linear. It is the names of the features that are associated
+            with each entry of the :meth:`coef_` parameter. Available only when the featurizer is not None and has
+            a method: `get_feature_names(feature_names)`. Otherwise None is returned.
+        """
+        if self._d_x is None:
+            # Handles the corner case when X=None but featurizer might be not None
+            return None
+        if feature_names is None:
+            feature_names = self._input_names["feature_names"]
+        if self.original_featurizer is None:
+            return feature_names
+        return get_feature_names_or_default(self.original_featurizer, feature_names)
+
+    @property
+    def model_final_(self):
+        # NOTE This is used by the inference methods and is more for internal use to the library
+        return self.ortho_learner_model_final_._model_final
+
+    @property
+    def model_cate(self):
+        """
+        Get the fitted final CATE model.
+
+        Returns
+        -------
+        model_cate: object of type(model_final)
+            An instance of the model_final object that was fitted after calling fit which corresponds
+            to the constant marginal CATE model.
+        """
+        return self.ortho_learner_model_final_._model_final
+
+    def shap_values(self, X, *, feature_names=None, treatment_names=None, output_names=None, background_samples=100):
+        return _shap_explain_model_cate(self.const_marginal_effect, self.model_cate, X, self._d_t, self._d_y,
+                                        featurizer=self.featurizer_,
+                                        feature_names=feature_names,
+                                        treatment_names=treatment_names,
+                                        output_names=output_names,
+                                        input_names=self._input_names,
+                                        background_samples=background_samples)
+    shap_values.__doc__ = LinearCateEstimator.shap_values.__doc__
+
+
+class DRIV(_BaseDRIV):
+    """
+    The DRIV algorithm for estimating CATE with IVs. It is the parent of the
+    public classes {LinearDRIV, SparseLinearDRIV,ForestDRIV}
+
+    Parameters
+    ----------
+    model_y_xw : estimator
```

Comment:
I think these docstrings should be mentioning what the "auto" behavior is, i.e. that we use Lasso or LogisticRegression

Reply:
Done